client/my-sites/email/email-management/home/email-plan.jsx

```diff
@@ -254,17 +256,47 @@ class EmailPlan extends React.Component {
 		);
 	}
 
-	render() {
-		const {
-			domain,
-			selectedSite,
-			hasSubscription,
-			purchase,
-			isLoadingPurchase,
-			translate,
-		} = this.props;
+	renderRenewalNavItem() {
+		const { purchase, selectedSite, translate } = this.props;
+
+		return (
+			<VerticalNavItem>
+				<RenewButton
+					purchase={ purchase }
+					selectedSite={ selectedSite }
+					subscriptionId={ parseInt( purchase.id, 10 ) }
+					tracksProps={ { source: 'email-plan-view' } }
+					customLabel={ translate( 'Renew now to add new mailboxes' ) }
+				/>
+			</VerticalNavItem>
+		);
+	}
+
+	renderAddNewMailboxOrRenewalNavItem() {
+		const { domain, hasSubscription, purchase, selectedSite, translate } = this.props;
+
+		if ( ! domain.currentUserCanManage || ! hasSubscription ) {
+			return null;
+		}
+
+		if ( ! selectedSite || ! purchase ) {
+			return <VerticalNavItem isPlaceholder />;
+		}
+
+		if ( needsToRenewSoon( purchase ) ) {
+			return this.renderRenewalNavItem();
+		}
 
 		const addMailboxProps = this.getAddMailboxProps();
```

Comment:
As far I can see `addMailboxProps` is only ever used once, so we could simply inline `this.getAddMailboxProps()`:

```javascript
			<VerticalNavItem { ...this.getAddMailboxProps() }>{ translate( 'Add new mailbox' ) }</VerticalNavItem>
```